### PR TITLE
gui: Fix #1300 - weighting % of devices according to folder size

### LINKED ISSUE #1300: In sync percentage should be weighted to folder size, not just average of folder completion
While syncing a folder of 26 GiB, 12.8GiB of which has been completed, Syncthing shows that as "Syncing (73%)"  at one end, while showing "Syncing (45%)" at the other end. In my case, the Documents folder is being synced. The "Default" folder exists, but has nothing of significant size to contribute. The 45% number looks more accurate to me. 

MacOS 10.10.1. 

syncing 73 percent
syncing 45 percent

### PR DESCRIPTION
**Pull request by 0x010C**
### Purpose

When 2 or more folders are shared with a remote device, the completion displayed in the device section on the right was based on a simple average of the completion of the shared folders. This is irrelevant when the folders have different sizes (e.g. a folder _X_ of 20MiB and a folder _Y_ of 100GiB; the _X_ folder will be quickly completely synchronised and bring the completion to 50%, even if just a very few part of the data has been send).

I weighted this average with the global size to reflect more accurately the progress of the synchronization.

### Testing

I have created two folders on an instance _A_ with very different sizes (1MiB and 2.3GiB). After asking to synchronised thooses two folders with an other device _B_, I checked regularly the percentage displayed on _A_ next-to the name of the device _B_.

### Authorship

This is my first contribution to syncthing.

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
There is an issue regarding this, which should be included in the commit title (fixes #xyz), also all the commits are prefixed with what they are fixing, etc `gui: Fix CSS`.
Furthermore you should add yourself to authors.

As for the actual fix, I'll leave for other to comment.

**Comment by calmh** [comment]
Looks reasonable enough, but I worry about rounding errors resulting in non-100% status when it should be etc. I think we should extend the folder completion event to include the byte counts directly instead.

**Comment by 0x010C** [comment]
@AudriusButkevicius Oh sorry for the issue, I didn't found it at my first look, and also apologize for the missing prefix. Should I close this pull-request and open a new one with a new commit?

For the authors file, it's written in the pull-request placeholder message that I shouldn't worry about it and that a senior contributor will do it; so I'm lost, should I do it or not?

**Comment by calmh** [comment]
Don't worry about the authors file, I'll add you before merging.

File: gui/default/syncthing/core/syncthingController.js [hunk 1]
```diff
@@ -312,15 +312,23 @@ angular.module('syncthing.core')
...
                 if (cmp === "_total") {
                     continue;
                 }
                 tot += $scope.completion[data.device][cmp] * $scope.model[cmp].globalBytes;
                 cnt += $scope.model[cmp].globalBytes;
+                if($scope.completion[data.device][cmp] != 100) {
+                    isComplete = false;
+                }
             }
             $scope.completion[data.device]._total = tot / cnt;
```
Comment (AudriusButkevicius) [inline gui/default/syncthing/core/syncthingController.js]: I'd move this in the else branch

File: gui/default/syncthing/core/syncthingController.js [hunk 2]
```diff
@@ -312,15 +312,23 @@ angular.module('syncthing.core')
...
-                cnt = 0;
+                cnt = 0,
+                isComplete = true;
             for (var cmp in $scope.completion[data.device]) {
                 if (cmp === "_total") {
                     continue;
                 }
                 tot += $scope.completion[data.device][cmp] * $scope.model[cmp].globalBytes;
                 cnt += $scope.model[cmp].globalBytes;
+                if($scope.completion[data.device][cmp] != 100) {
```
Comment (AudriusButkevicius) [inline gui/default/syncthing/core/syncthingController.js]: space after `if` everywhere pleasel

**Comment by 0x010C** [comment]
@AudriusButkevicius I just fixed both points you had pointed out.

**Comment by calmh** [comment]
@st-jenkins add to whitelist

**Comment by calmh** [comment]
@st-review lgtm, but i didn't test it

**Comment by st-review** [comment]
@calmh: Noted! Need another LGTM or explicit merge command.

**Comment by calmh** [comment]
@st-jenkins retest this please

**Comment by AudriusButkevicius** [comment]
@st-review lgtm

**Comment by st-review** [comment]
:ok_hand: Merged as c9cf01e0b6f80d4579a7a50c64339df91f0b1e4e. Thanks, @0x010C!

**Comment by AudriusButkevicius** [comment]
So this breaks the UI, due to a rate.

$scope.model[cmp].globalBytes is not always there when the calculation happens.

**Pull request closed without merging**